src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java

```diff
@@ -66,7 +66,12 @@ public static void addRunFacet(@Nonnull ContainerRecord record, @Nonnull Run<?,?
     /**
      * Creates a new {@link DockerFromFingerprintFacet} and adds a run. Or adds to an existing one.
      */
-    public static void addFromFacet(@Nonnull String imageId, @Nonnull Run<?,?> run) throws IOException {
+    public static void addFromFacet(@Nonnull String baseImageId, @Nonnull String derivedImageId, @Nonnull Run<?,?> run) throws IOException {
```

Comment:
This certainly deserves javadoc on the parameters.
baseImageId can be `null` if you create a [base](https://docs.docker.com/terms/image/#base-image) [image](https://docs.docker.com/articles/baseimages/)